pkg/api/kyverno/v1/types.go

```diff
@@ -120,34 +128,49 @@ type PolicyViolationList struct {
 type Policy struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
-	Spec              Spec         `json:"spec"`
-	Status            PolicyStatus `json:"status,omitempty"`
+	// Spec is the information to identify the policy
+	Spec Spec `json:"spec"`
+	// Status contains statistics related to policy
+	Status PolicyStatus `json:"status,omitempty"`
 }
 
 // Spec describes policy behavior by its rules
 type Spec struct {
-	Rules                   []Rule `json:"rules"`
+	// Rules contains the list of rules to be applied to resources
+	Rules []Rule `json:"rules"`
+	// ValidationFailureAction provides choice to enforce rules to resources during policy violations
 	ValidationFailureAction string `json:"validationFailureAction"`
-	Background              *bool  `json:"background"`
+	// Background provides choice for applying rules to existing resources
+	Background *bool `json:"background"`
 }
 
 // Rule is set of mutation, validation and generation actions
 // for the single resource description
 type Rule struct {
-	Name             string           `json:"name"`
-	MatchResources   MatchResources   `json:"match"`
+	// Name represents rule name
+	Name string `json:"name"`
+	// MatchResources contains resources for which the rule has to be applied
+	MatchResources MatchResources `json:"match"`
+	// ExcludeResources contains resources for which rule can be excluded
 	ExcludeResources ExcludeResources `json:"exclude,omitempty"`
```

Comment:
How are we indicating if a field is optional or required?

Reply:
Can we specify in the docs like this?

For required Field
// MatchResources - A required field contains resources for which the rule has to be applied.

For optional field
// ExcludeResources - An optional field contains resources for which rule can be excluded.

Reply:
Added required comments

Reply:
Can we simply add `Required` or `Optional` as the first word? See: https://htmlpreview.github.io/?https://github.com/kubernetes-sigs/wg-policy-prototypes/blob/master/policy-report/docs/index.html

Or, if we want to have compete sentences we should use this style:

**For required Field**
// MatchResources is a required field that contains resources for which the rule has to be applied.

**For optional field**
// ExcludeResources  is a optional field that contains resources for which rule can be excluded.

In the rendering, is the attribute name shown as the first column? In that case, it may be redundant to start the comments with the attribute name:

    "Attribute: <Attribute> is a ....". 

Instead we can shorten the help to something like this:

**For required Field**
// A required field that specifies when the rule should be applied.

**For optional field**
// An optional field that specifies when the rule should be excluded.


Or:

**For required Field**
// (Required.) Specifies when the rule should be applied.

**For optional field**
// (Optional.) Specifies when the rule should be excluded.


Or:

**For required Field**
// Required. Specifies when the rule should be applied.

**For optional field**
// Optional. Specifies when the rule should be excluded.

Reply:
yes @JimBugwadia , The Attribute name will be shown.
For reference [Kyverno API Docs](https://htmlpreview.github.io/?https://github.com/b-entangled/kyverno/blob/663_api_docs/documentation/index.html)

I think specifying 
**(Required)** or **(Optional)** as a prefix looks good.

Reply:
Can I modify it by specifying the prefixes?

Reply:
Yes, I like that as well! Also you can simply the text as suggested above.